### DISCUSSION TITLE
No module named "meshpy._internals"

### DISCUSSION THREAD

**Discussion by junfeng-chen** [inducer/meshpy#75] [Troubleshooting]
This module is imported in meshpy.triangle and meshpy.tet, but I cannot find it in meshpy. How come this import error? I checked the git, it does not exists either.

**Comment by inducer** [comment]
This issue tracker is for bugs, not technical support. Please send a message to the mailing list for tech support. One reason for this request is that I am typically the only one who sees issue email, and I am unable to handle the tech support load on my own.

**Comment by jwwang0-0** [comment]
Hi,

Sorry I am not sure if I should reopen this discussion but I got the same error. My meshpy version is 2020.1 with python3 and there is no mesh._internals exist. Do I miss anything to install?

**Reply by inducer** [reply]
How did you install?

If you're having installation/build trouble that you can't sort out yourself, I highly recommend that you install meshpy from conda-forge.

Grab miniforge from here: https://github.com/conda-forge/miniforge
then `conda install meshpy`.

**Reply by jwwang0-0** [reply]
Thank you for your quick response! Yes that is how exactly I installed it. My error is that there is no _internal file
![image](https://user-images.githubusercontent.com/91248123/162004209-e7b3e687-7a63-44d9-8267-c165b34fabe1.png)

**Reply by inducer** [reply]
What OS/architecture?

**Reply by jwwang0-0** [reply]
I am using Windows 10 and Python 3.8.12.

**Reply by inducer** [reply]
There's an `_internals` module in the 3.8+windows tarball here: https://anaconda.org/conda-forge/meshpy/files. Is that there on your filesystem?

**Reply by jwwang0-0** [reply]
Yes it is in the filesystem but not as a py file. Shall I rename it? 
![image](https://user-images.githubusercontent.com/91248123/162011444-6e10fccc-69b9-48b4-b369-7012e608fd80.png)

**Reply by jwwang0-0** [reply]
I googled about how to import a pyd file into python and did the following test and change pyd names but it is still not working

````python
import sys
sys.path.append("C:/Users/jwang/anaconda3/envs/fs2022/Lib/site-packages/meshpy")
import meshpy._internals as internals
````

**Reply by jwwang0-0** [reply]
Hi. Thank you for your help on this. It is figured out by my friend. The reason I cannot import it is because I am trying to use it in the ironpython environment (which is the python inside Rhino and Grasshopper, a modelling software in architectural industry). It supposes to work in normal python environment.

**Comment by samanthabeilman** [comment]
If there is a fix for this, I am looking for it after installing via pip. Sorry I am not super familiar with python. ':)

**Reply by samanthabeilman** [reply]
More info: I have also tried to install from the .tar file and the _internals file isn't there either.

**Comment by enilsson18** [comment]
Same issue, checked my local windows dependencies and they seemed fine but the pyd was still missing some. Seems like an issue with the pip installation on windows. Has anybody tried on a max/osx machine?
